File: Arrays/ContainerWithMaxWater.java
// Leetcode 11

class ContainerWithMaxArea {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length - 1;
        int maxArea = 0;

        while (i < j){
            int minHeight = Math.min(height[i], height[j]);
            int currArea = minHeight * (j - i);
            maxArea = Math.max(currArea, maxArea);

            if (height[i] < height[j]) i++;
            else j--;
        }

        return maxArea;
    }
}

File: Arrays/ContainerWithMaxWater.py
# Leetcode 11

class Solution(object):
    def maxArea(self, height):
        i, j = 0, len(height) - 1
        maxArea = 0

        while i < j:
            minHeight = min(height[i], height[j])
            currArea = minHeight * (j-i)
            maxArea = max(maxArea, currArea)
            
            if height[i] < height[j]: i += 1
            else: j -= 1

        return maxArea